esp-hal-common/src/reg_access.rs

```diff
@@ -0,0 +1,231 @@
+//! Utils
+//!
+//! # Overview
+//!
+//! Collection of struct which helps you write to registers.
+
+use crate::peripherals::generic::{Readable, Reg, RegisterSpec, Resettable, Writable};
+
+const U32_ALIGN_SIZE: usize = core::mem::size_of::<u32>();
+
+// ESP32 does reversed order
+#[cfg(esp32)]
+const U32_FROM_BYTES: fn([u8; 4]) -> u32 = u32::from_be_bytes;
+#[cfg(not(esp32))]
+const U32_FROM_BYTES: fn([u8; 4]) -> u32 = u32::from_ne_bytes;
+
+const U32_TO_BYTES: fn(u32) -> [u8; 4] = u32::to_ne_bytes;
```

Comment:
```suggestion
#[cfg(esp32)]
const U32_TO_BYTES: fn(u32) -> [u8; 4] = u32::to_be_bytes;
#[cfg(not(esp32))]
const U32_TO_BYTES: fn(u32) -> [u8; 4] = u32::to_ne_bytes;
```

It seems that `U32_FROM_BYTES` was updated, but `U32_TO_BYTES` was not. Making this change locally resolves the issue for ESP32.